bindings/c/test/mako/mako.cpp

```diff
@@ -1766,28 +1651,184 @@ void printReport(Arguments const& args,
+				fmt::fprintf(fp, "\"%s\": %lu", getOpName(op), point_99_9pct);
 			}
 		}
 	}
 	fmt::print("\n");
 	if (fp) {
 		fmt::fprintf(fp, "}}");
 	}
+}
+
+void printReport(Arguments& args, ThreadStatistics const* stats, double const duration_sec, pid_t pid_main, FILE* fp) {
+
+	auto final_stats = ThreadStatistics{};
+	const auto num_effective_threads = args.async_xacts > 0 ? args.async_xacts : args.num_threads;
+	for (auto i = 0; i < args.num_processes; i++) {
+		for (auto j = 0; j < num_effective_threads; j++) {
+			const auto idx = i * num_effective_threads + j;
+			final_stats.combine(stats[idx]);
+		}
+	}
+
+	/* overall stats */
+	fmt::printf("\n====== Total Duration %6.3f sec ======\n\n", duration_sec);
+	fmt::printf("Total Processes:   %8d\n", args.num_processes);
+	fmt::printf("Total Threads:     %8d\n", args.num_threads);
+	fmt::printf("Total Async Xacts: %8d\n", args.async_xacts);
+	if (args.tpsmax == args.tpsmin)
+		fmt::printf("Target TPS:        %8d\n", args.tpsmax);
+	else {
+		fmt::printf("Target TPS (MAX):  %8d\n", args.tpsmax);
+		fmt::printf("Target TPS (MIN):  %8d\n", args.tpsmin);
+		fmt::printf("TPS Interval:      %8d\n", args.tpsinterval);
+		fmt::printf("TPS Change:        ");
+		switch (args.tpschange) {
+		case TPS_SIN:
+			fmt::printf("%8s\n", "SIN");
+			break;
+		case TPS_SQUARE:
+			fmt::printf("%8s\n", "SQUARE");
+			break;
+		case TPS_PULSE:
+			fmt::printf("%8s\n", "PULSE");
+			break;
+		}
+	}
+	const auto tps_f = final_stats.getOpCount(OP_TRANSACTION) / duration_sec;
+	const auto tps_i = static_cast<uint64_t>(tps_f);
+	fmt::printf("Total Xacts:       %8lu\n", final_stats.getOpCount(OP_TRANSACTION));
+	fmt::printf("Total Conflicts:   %8lu\n", final_stats.getConflictCount());
+	fmt::printf("Total Errors:      %8lu\n", final_stats.getTotalErrorCount());
+	fmt::printf("Overall TPS:       %8lu\n\n", tps_i);
+
+	if (fp) {
+		fmt::fprintf(fp, "\"results\": {");
+		fmt::fprintf(fp, "\"totalDuration\": %6.3f,", duration_sec);
+		fmt::fprintf(fp, "\"totalProcesses\": %d,", args.num_processes);
+		fmt::fprintf(fp, "\"totalThreads\": %d,", args.num_threads);
+		fmt::fprintf(fp, "\"totalAsyncXacts\": %d,", args.async_xacts);
+		fmt::fprintf(fp, "\"targetTPS\": %d,", args.tpsmax);
+		fmt::fprintf(fp, "\"totalXacts\": %lu,", final_stats.getOpCount(OP_TRANSACTION));
+		fmt::fprintf(fp, "\"totalConflicts\": %lu,", final_stats.getConflictCount());
+		fmt::fprintf(fp, "\"totalErrors\": %lu,", final_stats.getTotalErrorCount());
+		fmt::fprintf(fp, "\"overallTPS\": %lu,", tps_i);
+	}
+
+	/* per-op stats */
+	printStatsHeader(args, true, true, false);
+
+	/* OPS */
+	putTitle("Total OPS");
+	if (fp) {
+		fmt::fprintf(fp, "\"totalOps\": {");
+	}
+	auto first_op = true;
+	for (auto op = 0; op < MAX_OP; op++) {
+		if ((args.txnspec.ops[op][OP_COUNT] > 0 && op != OP_TRANSACTION) || op == OP_COMMIT) {
+			putField(final_stats.getOpCount(op));
+			if (fp) {
+				if (first_op) {
+					first_op = 0;
```

Comment:
first op is a bool. Shouldn't `0` be `false`?

Reply:
Yes, it should. It's just the code copied from before. I'll change it to make it all consistent.